pkg/controllers/reallocation/utilization.go

```diff
@@ -43,11 +43,15 @@ func (u *Utilization) markUnderutilized(ctx context.Context, provisioner *v1alph
 
 	// 2. Get underutilized nodes
 	for _, node := range nodes {
+		// Don't consider nodes that are not ready
+		if !utilsnode.IsReady(*node) {
+			continue
+		}
 		pods, err := u.getPods(ctx, node)
 		if err != nil {
 			return fmt.Errorf("getting pods for node %s, %w", node.Name, err)
 		}
-		if utilsnode.IsEmpty(node, pods) {
+		if utilsnode.IsEmpty(pods) {
```

Comment:
It's a bit odd that utilsnode wouldn't take a node object. This helper maybe makes sense in scheduling or pods?

Reply:
Although the logic is related to the node itself, I agree that it's weird. I'll move it.